### PR TITLE
Aliens and monkeys can now climb structures

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
bugfix: Aliens can now open crates and lockers.
bugfix: Aliens and monkeys can now climb climbable objects, such as crates and sandbags.
rscadd: Aliens climb very, very fast.
/:cl:

### CODE REVIEW

File: code/game/objects/structures.dm [hunk 1]
```diff
@@ -92,8 +94,12 @@
...
+				structureclimber = null
+				return 0
 			density = 1
 			structureclimber = null
 			return 1
-	structureclimber = null
-	return ..()
+	else
+		structureclimber = null
+		return 0
```
Comment (phil235) [inline code/game/objects/structures.dm]:
```
	if(do_mob(user, user, adjusted_climb_time))
		 if(!qdeleted(src)) //Checking if structure has been destroyed
			density = 0
			if(step(user,get_dir(user,src.loc)))
				user.visible_message("<span class='warning'>[user] climbs onto [src].</span>", \
									"<span class='notice'>You climb onto [src].</span>")
				add_logs(user, src, "climbed onto")
				user.Stun(climb_stun)
				. = 1 //successful climb
			else
				user << "<span class='warning'>You fail to climb onto [src].</span>"
			density = 1
	structureclimber = null
```

You can remove the multiple "structureclimber = null" like this.

**Comment by GunHog** [comment]
Why not have climbable items respect the `PASSTABLE` flag? Things that fly or climb over tables with easy will be automatically accounted and kept consistent with tables. It helps with player learning as well.

**Comment by ChangelingRain** [comment]
Because projectiles use that flag, and climbable objects shouldn't necessarily respect that flag.

**Comment by lzimann** [comment]
can't monkeys simply go through tables anyway? and aliens can smash them with 1 click(and faster than climbing)

**Comment by Bawhoppen** [comment]
@lzimann can't do that with crates and sandbags

**Comment by lzimann** [comment]
oh i'm stupid, since i read but read it wrong

**Comment by KazeEspada** [comment]
Shouldn't monkeys climb fast too?

**Comment by ChangelingRain** [comment]
Monkeys are not fast, they're like little asshole humans.

**Pull request merged**